lib/Dancer2/Core/Request.pm

```diff
@@ -653,6 +653,14 @@ sub _shallow_clone {
     return $new_request;
 }
 
+
+sub route {
+    my ($self, $route) = @_;
+    if ($route) { $self->{route_matched} = $route; }
+
+    return $self->{route_matched};
```

Comment:
Hmm - we could tweak it so you can only set it once, if that would make you happier. 

I'd be worried, though, that that could cause problems if a request object is kept the same when a route fowards internally, and thus goes through the dispatcher a second time...

Or, I could make the `route()` method just an accessor, and have a private `_set_route()` setter?

I'm not quite sure I see the problem, to be honest?